provider/mesos.go

```diff
@@ -0,0 +1,449 @@
+package provider
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"crypto/tls"
+	"fmt"
+	"github.com/BurntSushi/ty/fun"
+	log "github.com/Sirupsen/logrus"
+	"github.com/cenkalti/backoff"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	"github.com/mesos/mesos-go/detector"
+	_ "github.com/mesos/mesos-go/detector/zoo" // Registers the ZK detector
+	"github.com/mesosphere/mesos-dns/detect"
+	"github.com/mesosphere/mesos-dns/logging"
+	"github.com/mesosphere/mesos-dns/records"
+	"github.com/mesosphere/mesos-dns/records/state"
+	"github.com/mesosphere/mesos-dns/util"
+	"sort"
+	"time"
+)
+
+//Mesos holds configuration of the mesos provider.
+type Mesos struct {
+	BaseProvider
+	Endpoint           string `description:"Marathon server endpoint. You can also specify multiple endpoint for Marathon"`
+	Domain             string `description:"Default domain used"`
+	ExposedByDefault   bool   `description:"Expose Marathon apps by default"`
+	GroupsAsSubDomains bool   `description:"Convert Mesos groups to subdomains"`
+	ZkDetectionTimeout int    `description:"ZkDetectionTimeout"`
+	RefreshSeconds     int    `description:"RefreshSeconds"`
+	IPSources          string `description:"IPSources"` // e.g. "host", "docker", "mesos", "rkt"
+	Masters            []string
```

Comment:
You should add the tag `description` on all configurable fields (if you want user to change those fields in TOML on flags).